MCM panics when a failed machine exists

Version: 0.12.0
For example, a VM is failed to be created. Aftwards, the following panic happens:

```
I0214 12:02:45.555090       1 machine.go:315] Creating machine shoot--core--alicloud-rf-worker-ky2cd-eu-central-1b-b685d7ndxzg, please wait!
I0214 12:02:45.592590       1 machine.go:514] Machine shoot--core--alicloud-rf-worker-ky2cd-eu-central-1b-b685d7lbnlx deleted successfully
panic: label name "failed_machine_name" missing in label map

goroutine 1252 [running]:
github.com/gardener/machine-controller-manager/vendor/github.com/prometheus/client_golang/prometheus.(*MetricVec).With(0xc00013cf80, 0xc00112aea0, 0xc000573480, 0x200f620)
    /tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/github.com/prometheus/client_golang/prometheus/vec.go:146 +0x79
github.com/gardener/machine-controller-manager/vendor/github.com/prometheus/client_golang/prometheus.(*GaugeVec).With(0xc000142068, 0xc00112aea0, 0x1dd403b, 0x17)
    /tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/github.com/prometheus/client_golang/prometheus/gauge.go:120 +0x38
github.com/gardener/machine-controller-manager/pkg/controller.(*controller).CollectMachineSetMetrics(0xc0003d0600, 0xc0000a8f00)
    /tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/metrics.go:226 +0x17b5
github.com/gardener/machine-controller-manager/pkg/controller.(*controller).Collect(0xc0003d0600, 0xc0000a8f00)
    /tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/metrics.go:327 +0x4d
github.com/gardener/machine-controller-manager/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func2(0xc00164f880, 0xc0000a8f00, 0x1fdcb20, 0xc0003d0600)
    /tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/github.com/prometheus/client_golang/prometheus/registry.go:382 +0x61
created by github.com/gardener/machine-controller-manager/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather
    /tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/github.com/prometheus/client_golang/prometheus/registry.go:380 +0x30
```

--- a/pkg/controller/metrics.go
+++ b/pkg/controller/metrics.go
@@ -18,12 +18,13 @@ limitations under the License.
 package controller
 
 import (
+	"strconv"
+
 	v1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
 	"github.com/gardener/machine-controller-manager/pkg/metrics"
 	"github.com/prometheus/client_golang/prometheus"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/labels"
-	"strconv"
 )
 
 // Describe is method required to implement the prometheus.Collect interface.
@@ -134,13 +135,13 @@ func (c *controller) CollectMachineDeploymentMetrics(ch chan<- prometheus.Metric
 		if machineDeployment.Status.FailedMachines != nil {
 			for _, failedMachine := range machineDeployment.Status.FailedMachines {
 				metrics.MachineDeploymentStatusFailedMachines.With(prometheus.Labels{
-					"name":                               mdMeta.Name,
-					"namespace":                          mdMeta.Namespace,
-					"failedMachine_name":                 failedMachine.Name,
-					"failedMachine_provider_id":          failedMachine.ProviderID,
-					"failedMachine_last_operation_state": string(failedMachine.LastOperation.State),
-					"failedMachine_last_operation_machine_operation_type": string(failedMachine.LastOperation.Type),
-					"failedMachine_owner_ref":                             failedMachine.OwnerRef}).Set(float64(1))
+					"name":                                mdMeta.Name,
+					"namespace":                           mdMeta.Namespace,
+					"failed_machine_name":                 failedMachine.Name,
+					"failed_machine_provider_id":          failedMachine.ProviderID,
+					"failed_machine_last_operation_state": string(failedMachine.LastOperation.State),
+					"failed_machine_last_operation_machine_operation_type": string(failedMachine.LastOperation.Type),
+					"failed_machine_owner_ref":                             failedMachine.OwnerRef}).Set(float64(1))
 
 			}
 		}
@@ -224,13 +225,13 @@ func (c *controller) CollectMachineSetMetrics(ch chan<- prometheus.Metric) {
 
 			for _, failedMachine := range *machineSet.Status.FailedMachines {
 				metrics.MachineSetStatusFailedMachines.With(prometheus.Labels{
-					"name":                               msMeta.Name,
-					"namespace":                          msMeta.Namespace,
-					"failedMachine_name":                 failedMachine.Name,
-					"failedMachine_provider_id":          failedMachine.ProviderID,
-					"failedMachine_last_operation_state": string(failedMachine.LastOperation.State),
-					"failedMachine_last_operation_machine_operation_type": string(failedMachine.LastOperation.Type),
-					"failedMachine_owner_ref":                             failedMachine.OwnerRef}).Set(float64(1))
+					"name":                                msMeta.Name,
+					"namespace":                           msMeta.Namespace,
+					"failed_machine_name":                 failedMachine.Name,
+					"failed_machine_provider_id":          failedMachine.ProviderID,
+					"failed_machine_last_operation_state": string(failedMachine.LastOperation.State),
+					"failed_machine_last_operation_machine_operation_type": string(failedMachine.LastOperation.Type),
+					"failed_machine_owner_ref":                             failedMachine.OwnerRef}).Set(float64(1))
 			}
 		}
 	}

